File: app/src/main/java/ua/in/quireg/anothermovieapp/movieList/SortOrderPicker.java
package ua.in.quireg.anothermovieapp.movieList;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.app.Dialog;
import android.content.DialogInterface;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;

import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

import ua.in.quireg.anothermovieapp.R;
import ua.in.quireg.anothermovieapp.common.Constants;

public class SortOrderPicker extends DialogFragment {

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle(R.string.ui_favourites_sort_order)
                .setItems(R.array.sort_orders_array, new DialogInterface.OnClickListener() {
                    @SuppressLint("CommitPrefEdits")
                    public void onClick(DialogInterface dialog, int which) {
                        SharedPreferences preferences =
                                PreferenceManager.getDefaultSharedPreferences(getContext());
                        preferences.edit().putInt(Constants.SORT_ORDER, which).apply();
                        getTargetFragment().onActivityResult(
                                getTargetRequestCode(), Activity.RESULT_OK,
                                getActivity().getIntent());
                    }
                });
        return builder.create();

    }
}

File: app/src/main/java/com/anothermovieapp/movieList/SortOrderPicker.kt
package com.anothermovieapp.movieList

import android.app.Activity
import android.app.Dialog
import android.os.Bundle
import android.preference.PreferenceManager
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.anothermovieapp.R
import com.anothermovieapp.common.Constants

class SortOrderPicker : DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity!!)
        builder.setTitle(R.string.ui_favourites_sort_order)
                .setItems(R.array.sort_orders_array) { dialog, which ->
                    val preferences = PreferenceManager.getDefaultSharedPreferences(context)
                    preferences.edit().putInt(Constants.SORT_ORDER, which).apply()
                    targetFragment!!.onActivityResult(
                            targetRequestCode, Activity.RESULT_OK,
                            activity!!.intent)
                }
        return builder.create()
    }
}